Upgrade PrimeNg-15-lte to PrimeNg 17

oidacra opened the pull request:
### Parent Issue
https://github.com/dotCMS/core/issues/29606

### Additonal Info
This PR is not for the master branch, but is for the **epic-primeng-upgrade** branch to break this epic into smaller PRs.

### Proposed Changes
* Upgrade PrimeNg-15-lte to PrimeNg 17

oidacra committed:
fix(core-web) changes

oidacra committed:
fix(core-web) changes

oidacra committed:
upgrade(chore) upgrade from PrimeNg15-LTW to PrimeNg 17

oidacra committed:
upgrade(chore) upgrade from PrimeNg15-LTW to PrimeNg 17

oidacra committed:
upgrade(chore) fix breadcrumb

oidacra committed:
upgrade(chore) fix breadcrumb

oidacra committed:
upgrade(chore) fix breadcrumb

oidacra committed:
upgrade(chore) fix breadcrumb

nicobytes committed:
chore(edit-content): delete treeselect clone

nicobytes committed:
chore(edit-content): delete treeselect clone

nicobytes committed:
chore(edit-content): apply format

nicobytes committed:
chore(edit-content): apply format

oidacra committed:
upgrade(chore) fix calendar

oidacra committed:
fix(core-web) last changes

fmontes commented on core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:
```diff
@@ -22,34 +22,33 @@ import { DotMessageService } from '@dotcms/data-access';
 import { DotPagesState, DotPageStore } from '../dot-pages-store/dot-pages.store';
 import { DotActionsMenuEventParams } from '../dot-pages.component';
 
+interface CustomContextMenu extends PrimeNGContextMenu {
+    show(event?: unknown): void;
+}
```
If this is a pure primeng component how come we can just change the type?

zJaaal replied:
Could be a PrimeNG issue

fmontes commented on core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:
```diff
@@ -22,34 +22,33 @@ import { DotMessageService } from '@dotcms/data-access';
         revision: this.dotMessageService.get('Revision'),
         draft: this.dotMessageService.get('Draft')
     };
+    private domIdMenuAttached = '';
+    private destroy$ = new Subject<boolean>();
+    private scrollElement?: HTMLElement;
```
```suggestion
    #domIdMenuAttached = '';
    #destroy$ = new Subject<boolean>();
    #scrollElement?: HTMLElement;
```

nicobytes replied:
Applied!

fmontes commented on core-web/apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts:
```diff
@@ -113,7 +114,12 @@ export class DotPageSelectorComponent implements ControlValueAccessor {
+                query: `//${site.hostname}/`,
+                originalEvent: {} as Event
+            };
+            // this.autoComplete.completeMethod.emit({ query: `//${site.hostname}/` });
```
Prob clean up, just adding it here so is not forgotted.

nicobytes replied:
Fxied!

Here's a breaking change: now `(onSelect)` doesn't send the selected item directly, instead it sends an obj with 2 values: originalEvent and value as selected item.

<img width="900" alt="Screenshot 2024-08-16 at 9 50 09 AM" src="https://github.com/user-attachments/assets/8b596631-e24f-4186-b1e1-b500e74c87e8">

fmontes commented on core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.ts:
```diff
@@ -77,14 +77,11 @@ export class DotListingDataTableComponent implements OnInit {
     dateColumns: DataTableColumn[];
     loading = true;
-    contextMenuItems: MenuItem[];
+    contextMenuItems: CustomMenuItem[];
```
Why we need a custom if we passing this to a primeng component?

nicobytes replied:
Fixed!

fmontes commented on core-web/apps/dotcms-ui/src/app/view/components/dot-portlet-base/components/dot-portlet-toolbar/dot-portlet-toolbar.component.ts:
```diff
@@ -37,7 +37,7 @@ export class DotPortletToolbarComponent {
      * @memberof DotPortletToolbarComponent
      */
-    onPrimaryClick($event: MouseEvent): void {
+    onPrimaryClick($event: unknown): void {
```
Why?

nicobytes replied:
Fixed!

fmontes commented on core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts:
```diff
@@ -1,6 +1,15 @@
+import { MenuItem, MenuItemCommandEvent } from 'primeng/api';
 
-export interface DotActionMenuItem {
+export interface CustomMenuItem<T = unknown> extends Omit<MenuItem, 'command'> {
+    command?(event?: T): void;
+}
```
We remove command with the Omit and then added back? we need to change the type in the function?

oidacra replied:
In a few cases, we call a command without parameter, and the lint fails.

fmontes commented on core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_autocomplete.scss:
```diff
@@ -90,7 +91,7 @@ p-autocomplete.p-inputwrapper-focus {
             &:active,
             &.p-focus {
-                border: none;
+                //border: none;
```
Remember clean up.

nicobytes replied:
Fixed!

oidacra committed:
fix(core-web) last changes

nicobytes committed:
save

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

oidacra committed:
fix(chore) merge from master

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

spbolton committed:
feat: Upgrade nx to 19.5.6  (#29415) (#29457)

Update nx version to the latest and make it consistent. This should
resolve issues causing #29415 (Flakey build. coreweb yarn install Bus
Error core ).

See https://nx.dev/nx-api/angular/documents/angular-nx-version-matrix

https://nx.dev/nx-api/workspace/documents/nx-nodejs-typescript-version-matrix
https://v17.angular.io/guide/versions

https://nx.dev/changelog

The main issue seems to be related to @nx/linter19.5.1 being added and
it not matching the rest of the nx modules with 18.3.5. Also separately
@nxext/stencil did not have a matching version to nx at 17.2.2. 17.x
versions of @nxext/stencil pull in other nx dependencies and needs an
exact version match, nx 18.x.x needs @nxext/stencil@18.0.0 and nx 19.x.x
needs @nxext/stencil@19.0.0

The conflicting versions showed up with "yarn list", nx versions can be
upgraded with the "nx migrate" command.

I have upgraded to the latest nx here after reviewing documents and any
breaking changes, but if there are problems preventing the nx update we
can try changing the nx version back to 18.3.5 and use a matching
version of @nx/linter as well as @nxext/stencil@18.0.0 to match

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
save

zJaaal commented on core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.stories.ts:
```diff
@@ -20,9 +22,7 @@ import {
 
-import { DotBlockEditorComponent } from './dot-block-editor.component';
-
-import { BlockEditorModule } from '../../block-editor.module';
+import { DotAiServiceMock } from '../../../../../data-access/src/lib/dot-ai/dot-ai-service.mock';
```
should we import this as `@dotcms/data-access`

nicobytes replied:
Fixed!

zJaaal commented on core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts:
```diff
@@ -1,6 +1,15 @@
+}
+
+export interface DotActionMenuItem<T = unknown> {
     shouldShow?: (x?: Record<string, unknown>) => boolean;
```
what is exactly `x`?

zJaaal commented on core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:
```diff
@@ -238,11 +244,12 @@ p-calendar.p-calendar-clearable {
     .p-calendar-clear-icon {
-        margin-top: -12px;
-        color: $color-palette-gray-500;
+        margin-top: -7px;
```
This value can change due to the font size update [PR](https://github.com/dotCMS/core/pull/29205)

nicobytes replied:
According to the storybook, it works fine!

<img width="459" alt="Screenshot 2024-08-16 at 10 11 46 AM" src="https://github.com/user-attachments/assets/d705f31e-4a53-483c-b793-65bfc418e068">

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

victoralfaro-dotcms committed:
Avoiding sending requests when there is no OpenAI API key to use.
Putting it in a centralized place.
Adjusting a couple of tests to reflect this logic change.
Also since from the provided logs there was an Jackson parsing error,
new fields were added to "DTO" class to consider the error field.
Initially when implementing IT for AI classes, I took the approach of
mocking or building an `AppConfig` class on the flight instead of
creating an `dotAI` app secrets so config classes can be created out of
them. Just as it happens at production code. I'm introducing some
changes to correct this as well.

nicobytes committed:
Merge branch 'master' of github.com:dotCMS/core

nicobytes committed:
chore: upgrade primeng

nicobytes committed:
chore: upgrade primeng

nicobytes committed:
chore: upgrade primeng

nicobytes committed:
chore: upgrade storybook

nicobytes committed:
chore: upgrade primeng

nicobytes committed:
Merge branch 'master' into issue-upgrade-primeng-to-17

nicobytes commented:
Based on the storybook, I find the following issues:

v15             |  v17
:-------------------------:|:-------------------------:
DotCMS/ThemeSelector | DotCMS/ThemeSelector
![Screenshot 2024-08-09 at 9 40 15 AM](https://github.com/user-attachments/assets/98f5efbc-5433-4435-b1b4-cb378c944d47) | ![Screenshot 2024-08-09 at 9 40 36 AM](https://github.com/user-attachments/assets/1e90472f-d9fb-4bc9-96e4-bad99d06a236)
DotCMS/Menu/Vertical | DotCMS/Forms/Vertical 
![Screenshot 2024-08-09 at 9 42 33 AM](https://github.com/user-attachments/assets/fe15eb5b-0ab7-4670-97e0-65a70d2304f3) | ![Screenshot 2024-08-09 at 9 42 44 AM](https://github.com/user-attachments/assets/34b1de11-d6ca-455b-b6b0-5189d114c24e)
DotCMS/Menu/DotCollapseBreadcrumb | DotCMS/Menu/DotCollapseBreadcrumb
![Screenshot 2024-08-09 at 9 52 25 AM](https://github.com/user-attachments/assets/f5d95fba-6f91-4ef0-9b66-d13caefba793)  | ![Screenshot 2024-08-09 at 9 52 38 AM](https://github.com/user-attachments/assets/8b4f706f-b75c-42d5-908e-0ad92505f8e0)
DotCMS/Menu/DotCrumbtrail | DotCMS/Menu/DotCrumbtrail
![Screenshot 2024-08-09 at 9 54 19 AM](https://github.com/user-attachments/assets/c32df8e6-3ecc-4209-b266-8c652e67b619) | ![Screenshot 2024-08-09 at 9 54 28 AM](https://github.com/user-attachments/assets/159e5326-caa5-489e-908b-80e3f3cf46b0)
PrimeNg/Button/SplitButton | PrimeNg/Button/SplitButton
![Screenshot 2024-08-09 at 9 56 26 AM](https://github.com/user-attachments/assets/4066dd9f-2005-4725-89e1-2edcd94fb871) | ![Screenshot 2024-08-09 at 9 56 38 AM](https://github.com/user-attachments/assets/694e7635-fed1-4202-a931-c8c3d8f40800)
![Screenshot 2024-08-09 at 9 57 44 AM](https://github.com/user-attachments/assets/d71e5cac-9444-421d-87c8-22451fa5afd8)  | ![Screenshot 2024-08-09 at 9 58 01 AM](https://github.com/user-attachments/assets/b15364ad-00d7-4dd5-b7b2-0325e9054bf6)
PrimeNg/Data/Table | PrimeNg/Data/Table
![Screenshot 2024-08-09 at 10 00 49 AM](https://github.com/user-attachments/assets/a3004e57-097a-4f9a-a7ac-5e6cc12f37ec) | ![Screenshot 2024-08-09 at 9 59 58 AM](https://github.com/user-attachments/assets/d27f1aff-c606-409f-9bad-b4cf1d7b1be8)
PrimeNg/Data/Tree | PrimeNg/Data/Tree
![Screenshot 2024-08-09 at 10 03 36 AM](https://github.com/user-attachments/assets/e0ea1e34-2774-424f-896d-4662df1ee5f8) | ![Screenshot 2024-08-09 at 10 03 22 AM](https://github.com/user-attachments/assets/ce0205d3-5d0a-430d-a4bd-7bc1fc4af7f2)
PrimeNg/Form| PrimeNg/Form
![Screenshot 2024-08-09 at 10 11 49 AM](https://github.com/user-attachments/assets/14ec3ed0-4217-4fdc-8ef4-7c34282b5517) | Not working
![Screenshot 2024-08-09 at 10 13 41 AM](https://github.com/user-attachments/assets/d073e27a-0b5e-4e19-8e96-a9a1ef1b056c)  |   ![Screenshot 2024-08-09 at 10 14 21 AM](https://github.com/user-attachments/assets/4f420b24-36f6-47b6-8df8-2452ca440c5e)
![Screenshot 2024-08-09 at 10 16 59 AM](https://github.com/user-attachments/assets/a49b764d-9dc1-416f-865e-fdcecac0a1ed) | Not working
![Screenshot 2024-08-09 at 10 19 00 AM](https://github.com/user-attachments/assets/eebbabfb-2806-4780-9860-c4845fe8e2be) | ![Screenshot 2024-08-09 at 10 19 12 AM](https://github.com/user-attachments/assets/f1e8d70d-4554-43a1-af29-b0921a491fe6)
![Screenshot 2024-08-09 at 10 20 35 AM](https://github.com/user-attachments/assets/81d8a705-aa05-4c8d-8c58-ab372201a0f1) | ![Screenshot 2024-08-09 at 10 20 52 AM](https://github.com/user-attachments/assets/f93729ab-4d23-45e1-88ec-4a8b269ada89) |
![Screenshot 2024-08-09 at 10 22 29 AM](https://github.com/user-attachments/assets/c52aa0e3-30d1-4bf8-b084-561772aa2919) | ![Screenshot 2024-08-09 at 10 22 44 AM](https://github.com/user-attachments/assets/efc2de13-8f64-470a-8531-df627c2498ba)
PrimeNg/Overlay | PrimeNg/Overlay
![Screenshot 2024-08-09 at 10 29 50 AM](https://github.com/user-attachments/assets/faf46138-7384-4183-b29b-90c780bf2c50)  |  ![Screenshot 2024-08-09 at 10 30 05 AM](https://github.com/user-attachments/assets/33bc2680-52f4-424a-a9a5-97310acd4faf)
![Screenshot 2024-08-09 at 10 31 00 AM](https://github.com/user-attachments/assets/fb6bbff1-a97d-4979-97b3-b3d5547e17fe) | ![Screenshot 2024-08-09 at 10 31 08 AM](https://github.com/user-attachments/assets/ce68e4d7-f86e-4655-af4a-6531bf759147)

nicobytes committed:
chore: upgrade storybook

nicobytes committed:
chore: upgrade storybook

nicobytes committed:
chore: upgrade storybook

nicobytes committed:
chore(primeng): fix error with p-dataview

nicobytes committed:
Merge branch 'master' into issue-upgrade-primeng-to-17

nicobytes committed:
chore(primeng): upgrade primeng

nicobytes committed:
chore(primeng): upgrade primeng

nicobytes committed:
chore(primeng): upgrade primeng

nicobytes committed:
Merge branch 'master' into issue-upgrade-primeng-to-17

nicobytes committed:
Merge branch '29606-upgrade-primeng-to-v17' into issue-upgrade-primeng-to-17

nicobytes committed:
Merge branch 'issue-upgrade-primeng-to-17' of github.com:dotCMS/core into issue-upgrade-primeng-to-17

nicobytes committed:
chore(primeng): fix erro with context menu

nicobytes committed:
chore(primeng): use native private attrs

nicobytes committed:
chore(primeng): fix onSelect with autocomplete

nicobytes committed:
chore(primeng): fix some errors

nicobytes committed:
chore(primeng): fix some errors

nicobytes committed:
Merge branch 'epic-primeng-upgrade' into issue-upgrade-primeng-to-17

nicobytes committed:
Merge branch 'epic-primeng-upgrade' into issue-upgrade-primeng-to-17

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: rollback

nicobytes committed:
chore: rollback

nicobytes committed:
Update CompletionsAPI.java

nicobytes committed:
Update utils.spec.ts

nicobytes committed:
Update dotcms-layout.component.ts

nicobytes committed:
Update CompletionsAPIImpl.java

nicobytes committed:
Update EmbeddingContentListenerTest.java

nicobytes committed:
Update graphql-page.js

nicobytes committed:
Update AIModelsTest.java

nicobytes committed:
Update AiTest.java

nicobytes committed:
Update dotAI.yml

nicobytes committed:
Update OpenAIRequest.java

nicobytes committed:
Update EmbeddingContentListener.java

nicobytes committed:
Update AIAppListener.java

nicobytes committed:
Update AppKeys.java

nicobytes committed:
Update AppConfig.java

nicobytes committed:
Update AIModels.java

nicobytes committed:
chore: fix errors

nicobytes committed:
Merge branch 'issue-upgrade-primeng-to-17' of github.com:dotCMS/core into issue-upgrade-primeng-to-17

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
chore: fix errors

nicobytes committed:
Merge branch 'epic-primeng-upgrade' into issue-upgrade-primeng-to-17

nicobytes committed:
chore: fix errors

nicobytes committed:
Merge branch 'issue-upgrade-primeng-to-17' of github.com:dotCMS/core into issue-upgrade-primeng-to-17
